cpp/include/raft/cudart_utils.h

```diff
@@ -259,11 +264,39 @@ void print_device_vector(const char* variable_name, const T* devMem,
 }
 /** @} */
 
-/** cuda malloc */
+static std::mutex mutex_;
+static std::unordered_map<void*, size_t> allocations;
+
+template <typename Type>
+void allocate(Type*& ptr, size_t len, rmm::cuda_stream_view stream,
+              bool setZero = false) {
+  size_t size = len * sizeof(Type);
+  ptr = (Type*)rmm::mr::get_current_device_resource()->allocate(size,
+                                                                stream.value());
```

Comment:
RMM's allocate function takes stream_view as an input argument, right? (so no need to call stream.value())